tools/wutool/wutool.cpp

```diff
@@ -1570,6 +1571,81 @@ class WuTool : public CppUnit::TestFixture
         ASSERT(numIterated == before);
     }
 
+    void testList3()
+    {
+        // Now by owner and 2 states via generic mechanism
+        Owned<IWorkUnitFactory> factory = getWorkUnitFactory();
+        unsigned start = msTick();
+        Owned<IConstWorkUnitIterator> wus;
+        __int64 cachehint = 0;
+        unsigned startRow = 0;
+
+        unsigned numCompleted = 0;
+        unsigned numFailed = 0;
+        WUSortField sortByOwner[] = { WUSFuser, WUSFstate, WUSFterm };
+
+        unsigned before = factory->numWorkUnits();
+        if (before > 0)
+        {
+            start = msTick();
+            wus.setown(factory->getWorkUnitsSorted((WUSortField) (WUSFwuid | WUSFreverse), sortByOwner, "WuTestUser00\0completed|failed", 0, 10000, NULL, NULL));
+            ForEach(*wus)
+            {
+                IConstWorkUnitInfo &wu = wus->query();
+                ASSERT(streq(wu.queryUser(), "WuTestUser00"));
+                ASSERT((wu.getState()==WUStateCompleted) || (wu.getState()==WUStateFailed));
+                if ((wu.getState()==WUStateCompleted))
+                    numCompleted++;
+                else
+                    numFailed++;
+            }
+            DBGLOG("%d completed workunits listed and %d failed workunits listed in %d ms", numCompleted, numFailed, msTick()-start);
+        }
+
+        if ((numCompleted > 0) || (numFailed > 0))
```

Comment:
You are correct and I should remove those two lines. My original thought assumes that a user will read the test result by line 1602). If the existing system does have both completed WUs and failed WUs, the user should know whether the numCompleted and numFailed are correct not from that result. In fact, I should not assumes that a user will read the test result.